### ISSUE TITLE
Bug: `rumdl check` succeeds, but `rumdl check --fix` exits non-success on the same workspace(without changing any files content between the runs)

### ISSUE THREAD

**Issue by pygarap** [rvben/rumdl#319]
### rumdl version

rumdl 0.0.222

### Installation method

Other

### Bug description

`rumdl check` reports success, but `rumdl check --fix` does not. On the same codebase and same command invocation, `--fix` appears to exit as a failure even though it reports zero issues.

### Steps to reproduce

❯ uvx rumdl@latest check

Success: No issues found in 68 files (32ms)

❯ uvx rumdl@latest check --fix

Issues: Found 0 issues in 10/68 files (148ms)

❯ uvx rumdl@latest check --fix --verbose

Enabled rules:
  - MD001 (Heading levels should only increment by one level at a time)
  - MD003 (Heading style)
  - MD005 (List indentation should be consistent)
  - MD007 (Unordered list indentation)
  - MD009 (Trailing spaces should be removed)
  - MD010 (No tabs)
  - MD011 (Reversed link syntax)
  - MD012 (Multiple consecutive blank lines)
  - MD014 (Commands in code blocks should show output)
  - MD018 (No space after hash in heading)
  - MD019 (Multiple spaces after hash in heading)
  - MD020 (No space inside hashes on closed heading)
  - MD021 (Multiple spaces inside hashes on closed heading)
  - MD022 (Headings should be surrounded by blank lines)
  - MD023 (Headings must start at the beginning of the line)
  - MD025 (Multiple top-level headings in the same document)
  - MD026 (Trailing punctuation in heading)
  - MD027 (Multiple spaces after quote marker (>))
  - MD028 (Blank line inside blockquote)
  - MD029 (Ordered list marker value)
  - MD034 (No bare URLs - wrap URLs in angle brackets)
  - MD035 (Horizontal rule style)
  - MD037 (Spaces inside emphasis markers)
  - MD038 (Spaces inside code span elements)
  - MD039 (Spaces inside link text)
  - MD040 (Code blocks should have a language specified)
  - MD042 (No empty links)
  - MD043 (Required heading structure)
  - MD044 (Proper names should have the correct capitalization)
  - MD045 (Images should have alternate text (alt text))
  - MD047 (Files should end with a single newline character)
  - MD048 (Code fence style should be consistent)
  - MD051 (Link fragments should reference valid headings)
  - MD052 (Reference links and images should use a reference that exists)
  - MD053 (Link and image reference definitions should be needed)
  - MD054 (Link and image style should be consistent)
  - MD055 (Table pipe style should be consistent)
  - MD056 (Table column count should be consistent)
  - MD057 (Relative links should point to existing files)
  - MD058 (Tables should be surrounded by blank lines)
  - MD059 (Link text should be descriptive)
  - MD061 (Forbidden terms)
  - MD062 (Link destination should not have leading or trailing whitespace)
  - MD063 (Heading capitalization)
  - MD064 (Multiple consecutive spaces)
  - MD065 (Horizontal rules should be surrounded by blank lines)
  - MD066 (Footnote validation)
  - MD067 (Footnote definitions should appear in order of first reference)
  - MD068 (Footnote definitions should not be empty)
  - MD069 (Duplicate list markers)
  - MD070 (Nested code fence collision - use longer fence to avoid premature closure)
  - MD071 (Blank line after frontmatter)
  - MD072 (Frontmatter keys should be sorted alphabetically)

Exclude patterns: ["site/**", ".venv/**", "dist/**", "build/**", ".git/**", ".vale/**", ".cache/**", "node_modules/**"]
Processing file: .github/ISSUE_TEMPLATE/bug_report.md
Processing file: docs/api/tags/tags-a-d.md
Processing file: docs/api/dependencies.md
Processing file: docs/learn/cookbook/static.md
Processing file: docs/sprint/index.md
Processing file: README.md
Processing file: docs/index.md
Processing file: CHANGELOG.md
Cache hit for /.github/ISSUE_TEMPLATE/bug_report.md
Processing file: CONTRIBUTING.md
Cache hit for /docs/api/tags/tags-a-d.md
Cache hit for /docs/learn/cookbook/static.md
Processing file: .github/ISSUE_TEMPLATE/feature_request.md
Cache hit for /docs/api/dependencies.md
Cache hit for /docs/sprint/index.md
Cache hit for /.github/ISSUE_TEMPLATE/feature_request.md
Cache hit for /docs/index.md
Cache hit for /CONTRIBUTING.md
Cache hit for /README.md
Cache hit for /CHANGELOG.md
Processing file: .github/ISSUE_TEMPLATE/doc_request.md
Processing file: docs/learn/escaping_html.md
Cache hit for /.github/ISSUE_TEMPLATE/doc_request.md
Cache hit for /docs/learn/escaping_html.md
Processing file: examples/airblog/posts/markdown-features.md
Cache hit for /examples/airblog/posts/markdown-features.md
Processing file: docs/api/exception_handlers.md
Cache hit for /docs/api/exception_handlers.md
Processing file: .github/pull_request_template.md
Cache hit for /.github/pull_request_template.md
Processing file: docs/api/exceptions.md
Cache hit for /docs/api/exceptions.md
Processing file: examples/containerize/README.md
Cache hit for /examples/containerize/README.md
Processing file: docs/api/tags/tags-e-m.md
Processing file: docs/api/forms.md
Cache hit for /docs/api/tags/tags-e-m.md
Cache hit for /docs/api/forms.md
Processing file: docs/api/index.md
Cache hit for /docs/api/index.md
Processing file: docs/api/tags/tags-n-s.md
Processing file: docs/learn/forms.md
Cache hit for /docs/api/tags/tags-n-s.md
Cache hit for /docs/learn/forms.md
Processing file: docs/api/layouts.md
Cache hit for /docs/api/layouts.md
Processing file: docs/api/tags/tags-t-z.md
Cache hit for /docs/api/tags/tags-t-z.md
Processing file: docs/api/middleware.md
Processing file: examples/airblog/README.md
Cache hit for /docs/api/middleware.md
Cache hit for /examples/airblog/README.md
Processing file: docs/api/models.md
Cache hit for /docs/api/models.md
Processing file: examples/deployment/k8s/README.md
Processing file: docs/api/templating.md
Cache hit for /docs/api/templating.md
Cache hit for /examples/deployment/k8s/README.md
Processing file: docs/api/requests.md
Processing file: docs/api/utils.md
Cache hit for /docs/api/requests.md
Cache hit for /docs/api/utils.md
Processing file: examples/airblog/posts/hello-world.md
Cache hit for /examples/airblog/posts/hello-world.md
Processing file: docs/community/index.md
Cache hit for /docs/community/index.md
Processing file: examples/src/README.md
Cache hit for /examples/src/README.md
Processing file: docs/sprint/src_examples.md
Processing file: docs/learn/air_tags.md
Cache hit for /docs/sprint/src_examples.md
Processing file: tests/benchmarks/README.md
Cache hit for /tests/benchmarks/README.md
Processing file: docs/learn/index.md
Cache hit for /docs/learn/index.md
Processing file: docs/api/responses.md
Cache hit for /docs/learn/air_tags.md
Cache hit for /docs/api/responses.md
Processing file: docs/api/routing.md
Processing file: CODE_OF_CONDUCT.md
Cache hit for /docs/api/routing.md
Processing file: docs/about/roadmap.md
Cache hit for /CODE_OF_CONDUCT.md
Cache hit for /docs/about/roadmap.md
Processing file: docs/about/why.md
Processing file: docs/community/tools.md
Cache hit for /docs/community/tools.md
Processing file: MAINTAINERS.md
Cache hit for /docs/about/why.md
Processing file: docs/api/svg/index.md
Cache hit for /MAINTAINERS.md
Cache hit for /docs/api/svg/index.md
Processing file: docs/learn/jinja.md
Processing file: docs/api/svg/svg-tags-a-d.md
Cache hit for /docs/learn/jinja.md
Cache hit for /docs/api/svg/svg-tags-a-d.md
Processing file: docs/learn/cookbook/charts.md
Cache hit for /docs/learn/cookbook/charts.md
Processing file: docs/api/svg/svg-tags-e-m.md
Cache hit for /docs/api/svg/svg-tags-e-m.md
Processing file: docs/about/story.md
Cache hit for /docs/about/story.md
Processing file: docs/learn/cookbook/forms.md
Cache hit for /docs/learn/cookbook/forms.md
Processing file: docs/api/svg/svg-tags-n-s.md
Cache hit for /docs/api/svg/svg-tags-n-s.md
Processing file: docs/learn/cookbook/index.md
Cache hit for /docs/learn/cookbook/index.md
Processing file: docs/api/applications.md
Cache hit for /docs/api/applications.md
Processing file: docs/learn/layouts.md
Cache hit for /docs/learn/layouts.md
Processing file: docs/api/background.md
Cache hit for /docs/api/background.md
Processing file: docs/learn/quickstart.md
Processing file: docs/about/help_air.md
Processing file: docs/api/svg/svg-tags-t-z.md
Processing file: docs/learn/cookbook/minimal.md
Cache hit for /docs/about/help_air.md
Cache hit for /docs/api/svg/svg-tags-t-z.md
Cache hit for /docs/learn/cookbook/minimal.md
Processing file: docs/api/tags/index.md
Cache hit for /docs/api/tags/index.md
Processing file: docs/about/index.md
Cache hit for /docs/about/index.md
Cache hit for /docs/learn/quickstart.md
Processing file: docs/learn/cookbook/page-decorator.md
Processing file: docs/learn/sqlmodel.md
Cache hit for /docs/learn/cookbook/page-decorator.md
Cache hit for /docs/learn/sqlmodel.md
Processing file: docs/about/inspirations.md
Processing file: docs/learn/cookbook/authentication.md
Cache hit for /docs/about/inspirations.md
Cache hit for /docs/learn/cookbook/authentication.md
Processing file: docs/learn/cookbook/bigger-applications.md
Cache hit for /docs/learn/cookbook/bigger-applications.md
Loaded workspace index from cache with 68 files
Workspace index: 0 updated, 68 unchanged, 0 pruned (68 total) in 1.523917ms
Saved workspace index cache with 68 files

Issues: Found 0 issues in 10/68 files (157ms)

### Expected behavior

* If there are no issues, `rumdl check --fix` should behave like `rumdl check`:

  * report "no issues"
  * process the full workspace consistently (or clearly explain why it doesn't)
  * exit with a success code

### Actual behavior

* `rumdl check`:

  * reports success for 68 files
* `rumdl check --fix`:

  * reports `Found 0 issues in 10/68 files`
  * appears to fail (non-success) despite reporting zero issues

### Configuration

```toml

```

### Operating system

macOS

### Additional context

* The output difference (`10/68` vs `68`) is confusing and makes it hard to use `--fix` in CI.
* `--verbose` shows many cache hits and ends with "Found 0 issues", so it is not clear why the command is not treated as success.

**Comment by rvben** [comment]
Fixed in v0.0.223!

The issue was that the re-lint pass in fix mode wasn't applying the same per-file-ignores and inline config filtering as the original lint, which caused inconsistent file counting and exit behavior.

The fix ensures that:
- `rumdl check` and `rumdl check --fix` now produce consistent output when there are no issues
- Exit code 0 is returned when all violations are fixed (or when there are none)
- The "X/Y files" count correctly reflects files with actual issues, not just processed files

Thanks @pygarap for the detailed bug report with verbose output - it helped identify the root cause!
